src/metabase/models/table.clj

```diff
@@ -62,9 +62,10 @@
                   :field_order  (driver/default-field-order (t2/select-one-fn :engine Database :id (:db_id table)))}]
     (merge defaults table)))
 
+;; TODO: should we do this after the deletion?
 (t2/define-before-delete :model/Table
   [{:keys [db_id schema id]}]
-  (t2/delete! Permissions :object [:like (str (perms/data-perms-path db_id schema id) "%")]))
+  (t2/delete! Permissions :object [:like (str "%" (perms/data-perms-path db_id schema id) "%")]))
```

Comment:
The leading `%` gives me pause, but I defer to @noahmoss

Reply:
We need the `%` because of prefixes like `download` or `data-model`. 
https://github.com/metabase/metabase/blob/d35c779976ae6bd79369e9cc6bec8540908f37d2/src/metabase/models/permissions.clj#L257

This is covered by ngoc's tests

Reply:
Yeah this should be fine—we should have had it work like that already.